### ISSUE TITLE
[AIR SDK 51.1.2.2][macOS] macOS application with ANE packaging broken

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#3562]
## Problem Description
macOS application which use ANE packaging broken with AIR SDK 51.1.2.2.

Tested with AIR SDK 51.1.2.2 with different macOS devices with applications which use multiple ANEs (some of them with frameworks). Native extensions built using AIR 51.1.1.5.
Tested with and without `UseNativeCodesign=true`.
There is no such issue if application didn't use ANE.
There is no such issue using AIR SDK 51.1.2.1 or below.

Related issues:
https://github.com/airsdk/Adobe-Runtime-Support/issues/3390
https://github.com/airsdk/Adobe-Runtime-Support/discussions/2088
https://github.com/airsdk/Adobe-Runtime-Support/issues/2083
https://github.com/airsdk/Adobe-Runtime-Support/issues/1090

## Steps to Reproduce
Try to package AIR application for macOS which uses ANEs with AIR SDK 51.1.2.2.
Call `adt -package ...` command.

**Actual Result:**
Packaging error:
```
Failure (1) in codesign request.
/Users/username/builds/build/adt15380117071026662038.tmp/temp: replacing existing signature
/Users/username/builds/build/adt15380117071026662038.tmp/temp: code object is not signed at all
In subcomponent: /Users/username/builds/build/adt15380117071026662038.tmp/temp/Contents/Frameworks/MyANE2.framework
```
In adt.log you can see:
```
Called by: /Users/username/Documents/AIR_SDK/lib/adt.jar
ADT 51.1.2.2 called with: -package -storetype pkcs12 -keystore /Users/username/Documents/cert.p12 -storepass xxxxxx -keypass xxxxxx -target bundle -storetype KeychainStore -alias Developer ID Application: Company Name (ID) MyApp.app descriptor.xml -extdir /Users/username/builds/libs -extdir /Users/username/builds/ane-all -extdir /Users/username/builds/ane-full MyApp.swf
Creating temporary AIR package at /Users/username/builds/build/air2401025436446791822.tmp
Converting AIR file:  /Users/username/Documents/AIR_SDK/lib/nai/bin/naip -convert /Users/username/builds/build/air2401025436446791822.tmp /Users/username/builds/build/adt3172276520321907400.tmp/temp
Renaming ANE framework (MyANE1.framework) = successRenaming ANE framework (MyANE2.framework) = successCodesign request: find /Users/username/builds/build/adt3172276520321907400.tmp/temp -name "*.framework" -exec codesign -s 'Developer ID Application: Company Name (ID)' -f -o runtime --strict --timestamp -v {} ;
Codesign request: find /Users/username/builds/build/adt3172276520321907400.tmp/temp -name "*.plugin" -exec codesign -s 'Developer ID Application: Company Name (ID)' -f -o runtime --strict --timestamp -v {} ;
Codesign request: find /Users/username/builds/build/adt3172276520321907400.tmp/temp -name "*.dylib" -exec codesign -s 'Developer ID Application: Company Name (ID)' -f -o runtime --strict --timestamp -v {} ;
Codesign request: find /Users/username/builds/build/adt3172276520321907400.tmp/temp -name "A2712Enabler" -exec codesign --entitlements /var/folders/2w/4v61661n3yv3w94_gft9_h5w0000gq/T/entitlement1610265318186720629.tmp -s 'Developer ID Application: Company Name (ID)' -f -o runtime --strict --timestamp -v {} ;
Codesign request: codesign --entitlements /var/folders/2w/4v61661n3yv3w94_gft9_h5w0000gq/T/entitlement1610265318186720629.tmp -s Developer ID Application: Company Name (ID) -f -o runtime --strict --timestamp -v /Users/username/builds/build/adt3172276520321907400.tmp/temp
/Users/username/builds/build/adt3172276520321907400.tmp/temp: replacing existing signature
/Users/username/builds/build/adt3172276520321907400.tmp/temp: code object is not signed at all
In subcomponent: /Users/username/builds/build/adt3172276520321907400.tmp/temp/Contents/Frameworks/MyANE2.framework
Process response code = 1
```

**Expected Result:**
Application packaged without errors.

## Known Workarounds
none
*use AIR SDK 51.1.2.1 or below

**Comment by ajwfrost** [comment]
Curiously - the change here was to fix things for builds that were intended for App Store, and built on older macOS versions, where we were including entitlements into the frameworks (but that's not allowed for App Store apps.. not sure why it worked for DevID apps..). But revising it then exposed an issue which then makes it dependent upon the actual ANEs - and whether their frameworks are already signed.

We've got a fix for this, will be isolated to just adt.jar so fairly straightforward, will get it released shortly,.

thanks

**Comment by a9848840** [comment]
Hi @ajwfrost 

AIR 51.1.1.5 ipa packaged successfully

But AIR 51.1.2.2
Encountered when packaging ipa on mac

I'm looking forward to the next Air version

thx

> sh packageapp.sh
> Undefined symbols for architecture arm64:
>   "_$s4Body7SwiftUI12ViewModifierPTl", referenced from:
>       l_got.$s4Body7SwiftUI12ViewModifierPTl in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierMp", referenced from:
>       l_got.$s7SwiftUI12ViewModifierMp in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierP05_makeC08modifier6inputs4bodyAA01_C7OutputsVAA11_GraphValueVyxG_AA01_C6InputsVAiA01_J0V_ANtctFZTq", referenced from:
>       l_got.$s7SwiftUI12ViewModifierP05_makeC08modifier6inputs4bodyAA01_C7OutputsVAA11_GraphValueVyxG_AA01_C6InputsVAiA01_J0V_ANtctFZTq in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierP05_makeC4List8modifier6inputs4bodyAA01_cF7OutputsVAA11_GraphValueVyxG_AA01_cF6InputsVAiA01_K0V_ANtctFZTq", referenced from:
>       l_got.$s7SwiftUI12ViewModifierP05_makeC4List8modifier6inputs4bodyAA01_cF7OutputsVAA11_GraphValueVyxG_AA01_cF6InputsVAiA01_K0V_ANtctFZTq in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierP14_viewListCount6inputs4bodySiSgAA01_cfG6InputsV_AgIXEtFZTq", referenced from:
>       l_got.$s7SwiftUI12ViewModifierP14_viewListCount6inputs4bodySiSgAA01_cfG6InputsV_AgIXEtFZTq in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierP4BodyAC_AA0C0Tn", referenced from:
>       l_got.$s7SwiftUI12ViewModifierP4BodyAC_AA0C0Tn in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierP4body7content4BodyQzAA01_cD8_ContentVyxG_tFTq", referenced from:
>       l_got.$s7SwiftUI12ViewModifierP4body7content4BodyQzAA01_cD8_ContentVyxG_tFTq in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierPAAE05_makeC08modifier6inputs4bodyAA01_C7OutputsVAA11_GraphValueVyxG_AA01_C6InputsVAiA01_J0V_ANtctFZ", referenced from:
>       _$s17FirebaseAnalytics06LoggedB8ModifierV7SwiftUI04ViewD0AadEP05_makeG08modifier6inputs4bodyAD01_G7OutputsVAD11_GraphValueVyxG_AD01_G6InputsVAlD01_M0V_AQtctFZTW in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierPAAE05_makeC4List8modifier6inputs4bodyAA01_cF7OutputsVAA11_GraphValueVyxG_AA01_cF6InputsVAiA01_K0V_ANtctFZ", referenced from:
>       _$s17FirebaseAnalytics06LoggedB8ModifierV7SwiftUI04ViewD0AadEP05_makeG4List8modifier6inputs4bodyAD01_gI7OutputsVAD11_GraphValueVyxG_AD01_gI6InputsVAlD01_N0V_AQtctFZTW in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierPAAE14_viewListCount6inputs4bodySiSgAA01_cfG6InputsV_AgIXEtFZ", referenced from:
>       _$s17FirebaseAnalytics06LoggedB8ModifierV7SwiftUI04ViewD0AadEP14_viewListCount6inputs4bodySiSgAD01_giJ6InputsV_AjLXEtFZTW in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI15ModifiedContentVMa", referenced from:
>       l_get_witness_table 7SwiftUI4ViewRzlAA15ModifiedContentVyx17FirebaseAnalytics06LoggedG8ModifierVGAaBHPxAaBHD1__AgA0cI0HPyHCHC in FirebaseAnalytics(Analytics+SwiftUI.o)
>       _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGMa in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI15ModifiedContentVMn", referenced from:
>       _symbolic _____yx_____G 7SwiftUI15ModifiedContentV 17FirebaseAnalytics06LoggedF8ModifierV in FirebaseAnalytics(Analytics+SwiftUI.o)
>       _symbolic _____y_____y_____G_____G 7SwiftUI15ModifiedContentV AA014_ViewModifier_D0V 17FirebaseAnalytics06LoggedhF0V AA017_AppearanceActionF0V in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI15ModifiedContentVyxq_GAA4ViewA2aERzAA0E8ModifierR_rlMc", referenced from:
>       l_get_witness_table 7SwiftUI4ViewRzlAA15ModifiedContentVyx17FirebaseAnalytics06LoggedG8ModifierVGAaBHPxAaBHD1__AgA0cI0HPyHCHC in FirebaseAnalytics(Analytics+SwiftUI.o)
>       _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGACyxq_GAA0E0A2aNRzAA0eF0R_rlWl in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI21_ViewModifier_ContentVMa", referenced from:
>       _$s7SwiftUI21_ViewModifier_ContentVy17FirebaseAnalytics06LoggedgD0VGMa in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI21_ViewModifier_ContentVMn", referenced from:
>       _symbolic _____y_____y_____G_____G 7SwiftUI15ModifiedContentV AA014_ViewModifier_D0V 17FirebaseAnalytics06LoggedhF0V AA017_AppearanceActionF0V in FirebaseAnalytics(Analytics+SwiftUI.o)
>       _symbolic _____y_____G 7SwiftUI21_ViewModifier_ContentV 17FirebaseAnalytics06LoggedgD0V in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI21_ViewModifier_ContentVyxGAA0C0AAMc", referenced from:
>       _$s7SwiftUI21_ViewModifier_ContentVy17FirebaseAnalytics06LoggedgD0VGACyxGAA0C0AAWl in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI25_AppearanceActionModifierVAA04ViewE0AAWP", referenced from:
>       _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGACyxq_GAA0E0A2aNRzAA0eF0R_rlWl in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI25_AppearanceActionModifierVMn", referenced from:
>       _symbolic _____y_____y_____G_____G 7SwiftUI15ModifiedContentV AA014_ViewModifier_D0V 17FirebaseAnalytics06LoggedhF0V AA017_AppearanceActionF0V in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI25_AppearanceActionModifierVN", referenced from:
>       _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGMa in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI4ViewMp", referenced from:
>       l_got.$s7SwiftUI4ViewMp in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI4ViewPAAE8modifieryAA15ModifiedContentVyxqd__Gqd__lF", referenced from:
>       _$s7SwiftUI4ViewP17FirebaseAnalyticsE15analyticsScreen4name5class15extraParametersQrSS_SSSDySSypGtF in FirebaseAnalytics(Analytics+SwiftUI.o)
> ld: symbol(s) not found for architecture arm64
> 
> Compilation failed while executing : ld64

**Comment by a9848840** [comment]
Thank you, ajwfrost
It turns out to be FirebaseAnalytics
:(

**Comment by Seraphim-AT** [comment]
Hey! 

For the `Undefined symbols for architecture arm64` issue you can try to add iOS SDK as mentioned here, it worked for me with AIR 51.1.2.2:

https://docs.airnativeextensions.com/docs/adverts/troubleshooting/#ios-sdk-issue

> Hi @ajwfrost
> 
> AIR 51.1.1.5 ipa packaged successfully
> 
> But AIR 51.1.2.2 Encountered when packaging ipa on mac
> 
> I'm looking forward to the next Air version
> 
> thx
> 
> > sh packageapp.sh
> > Undefined symbols for architecture arm64:
> > "_$s4Body7SwiftUI12ViewModifierPTl", referenced from:
> > l_got.$s4Body7SwiftUI12ViewModifierPTl in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierMp", referenced from:
> > l_got.$s7SwiftUI12ViewModifierMp in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierP05_makeC08modifier6inputs4bodyAA01_C7OutputsVAA11_GraphValueVyxG_AA01_C6InputsVAiA01_J0V_ANtctFZTq", referenced from:
> > l_got.$s7SwiftUI12ViewModifierP05_makeC08modifier6inputs4bodyAA01_C7OutputsVAA11_GraphValueVyxG_AA01_C6InputsVAiA01_J0V_ANtctFZTq in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierP05_makeC4List8modifier6inputs4bodyAA01_cF7OutputsVAA11_GraphValueVyxG_AA01_cF6InputsVAiA01_K0V_ANtctFZTq", referenced from:
> > l_got.$s7SwiftUI12ViewModifierP05_makeC4List8modifier6inputs4bodyAA01_cF7OutputsVAA11_GraphValueVyxG_AA01_cF6InputsVAiA01_K0V_ANtctFZTq in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierP14_viewListCount6inputs4bodySiSgAA01_cfG6InputsV_AgIXEtFZTq", referenced from:
> > l_got.$s7SwiftUI12ViewModifierP14_viewListCount6inputs4bodySiSgAA01_cfG6InputsV_AgIXEtFZTq in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierP4BodyAC_AA0C0Tn", referenced from:
> > l_got.$s7SwiftUI12ViewModifierP4BodyAC_AA0C0Tn in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierP4body7content4BodyQzAA01_cD8_ContentVyxG_tFTq", referenced from:
> > l_got.$s7SwiftUI12ViewModifierP4body7content4BodyQzAA01_cD8_ContentVyxG_tFTq in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierPAAE05_makeC08modifier6inputs4bodyAA01_C7OutputsVAA11_GraphValueVyxG_AA01_C6InputsVAiA01_J0V_ANtctFZ", referenced from:
> > _$s17FirebaseAnalytics06LoggedB8ModifierV7SwiftUI04ViewD0AadEP05_makeG08modifier6inputs4bodyAD01_G7OutputsVAD11_GraphValueVyxG_AD01_G6InputsVAlD01_M0V_AQtctFZTW in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierPAAE05_makeC4List8modifier6inputs4bodyAA01_cF7OutputsVAA11_GraphValueVyxG_AA01_cF6InputsVAiA01_K0V_ANtctFZ", referenced from:
> > _$s17FirebaseAnalytics06LoggedB8ModifierV7SwiftUI04ViewD0AadEP05_makeG4List8modifier6inputs4bodyAD01_gI7OutputsVAD11_GraphValueVyxG_AD01_gI6InputsVAlD01_N0V_AQtctFZTW in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI12ViewModifierPAAE14_viewListCount6inputs4bodySiSgAA01_cfG6InputsV_AgIXEtFZ", referenced from:
> > _$s17FirebaseAnalytics06LoggedB8ModifierV7SwiftUI04ViewD0AadEP14_viewListCount6inputs4bodySiSgAD01_giJ6InputsV_AjLXEtFZTW in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI15ModifiedContentVMa", referenced from:
> > l_get_witness_table 7SwiftUI4ViewRzlAA15ModifiedContentVyx17FirebaseAnalytics06LoggedG8ModifierVGAaBHPxAaBHD1__AgA0cI0HPyHCHC in FirebaseAnalytics(Analytics+SwiftUI.o)
> > _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGMa in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI15ModifiedContentVMn", referenced from:
> > _symbolic _____yx_____G 7SwiftUI15ModifiedContentV 17FirebaseAnalytics06LoggedF8ModifierV in FirebaseAnalytics(Analytics+SwiftUI.o)
> > _symbolic _____y_____y_____G_____G 7SwiftUI15ModifiedContentV AA014_ViewModifier_D0V 17FirebaseAnalytics06LoggedhF0V AA017_AppearanceActionF0V in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI15ModifiedContentVyxq_GAA4ViewA2aERzAA0E8ModifierR_rlMc", referenced from:
> > l_get_witness_table 7SwiftUI4ViewRzlAA15ModifiedContentVyx17FirebaseAnalytics06LoggedG8ModifierVGAaBHPxAaBHD1__AgA0cI0HPyHCHC in FirebaseAnalytics(Analytics+SwiftUI.o)
> > _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGACyxq_GAA0E0A2aNRzAA0eF0R_rlWl in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI21_ViewModifier_ContentVMa", referenced from:
> > _$s7SwiftUI21_ViewModifier_ContentVy17FirebaseAnalytics06LoggedgD0VGMa in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI21_ViewModifier_ContentVMn", referenced from:
> > _symbolic _____y_____y_____G_____G 7SwiftUI15ModifiedContentV AA014_ViewModifier_D0V 17FirebaseAnalytics06LoggedhF0V AA017_AppearanceActionF0V in FirebaseAnalytics(Analytics+SwiftUI.o)
> > _symbolic _____y_____G 7SwiftUI21_ViewModifier_ContentV 17FirebaseAnalytics06LoggedgD0V in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI21_ViewModifier_ContentVyxGAA0C0AAMc", referenced from:
> > _$s7SwiftUI21_ViewModifier_ContentVy17FirebaseAnalytics06LoggedgD0VGACyxGAA0C0AAWl in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI25_AppearanceActionModifierVAA04ViewE0AAWP", referenced from:
> > _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGACyxq_GAA0E0A2aNRzAA0eF0R_rlWl in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI25_AppearanceActionModifierVMn", referenced from:
> > _symbolic _____y_____y_____G_____G 7SwiftUI15ModifiedContentV AA014_ViewModifier_D0V 17FirebaseAnalytics06LoggedhF0V AA017_AppearanceActionF0V in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI25_AppearanceActionModifierVN", referenced from:
> > _$s7SwiftUI15ModifiedContentVyAA014_ViewModifier_D0Vy17FirebaseAnalytics06LoggedhF0VGAA017_AppearanceActionF0VGMa in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI4ViewMp", referenced from:
> > l_got.$s7SwiftUI4ViewMp in FirebaseAnalytics(Analytics+SwiftUI.o)
> > "_$s7SwiftUI4ViewPAAE8modifieryAA15ModifiedContentVyxqd__Gqd__lF", referenced from:
> > _$s7SwiftUI4ViewP17FirebaseAnalyticsE15analyticsScreen4name5class15extraParametersQrSS_SSSDySSypGtF in FirebaseAnalytics(Analytics+SwiftUI.o)
> > ld: symbol(s) not found for architecture arm64
> > Compilation failed while executing : ld64

**Comment by a9848840** [comment]
Hi
@Seraphim-AT 

I saw 3 people reporting the same issue in the ANE-Firebase report area.

I will try add iOS SDK in mac packageapp, thanks for your suggestion :)

**Comment by marchbold** [comment]
@a9848840 In our tests linking against the iOS SDK resolves this issue. Should give you a way forward while we look into the issue.

**Comment by a9848840** [comment]
Hi
@marchbold 
I tried to add IOS SDK, but the result failed, asking how to add IOS SDK
IOS SDK --- > iPhoneOS17.5.sdk

Is it because the iOS SDK is too old?
Are there any new loading points?
Thanks
================================

> Undefined symbols for architecture arm64:
>   "_$s4Body7SwiftUI12ViewModifierPTl", referenced from:
>       l_got.$s4Body7SwiftUI12ViewModifierPTl in FirebaseAnalytics(Analytics+SwiftUI.o)
>   "_$s7SwiftUI12ViewModifierMp", referenced from:..........

================================

# P12
KEY_P12="cert/iOS/testfly/******.p12"

# Password
KEY_PASSWORD=

# Profile
PROVISIONING_PROFILE="cert/iOS/testfly/********.mobileprovision"

# Xml
APP_XML="application_ios_testfight.xml"

# Ipa
IPA_FILE="dist/*******.ipa"

# Command
PACKAGE_COMMAND="ipa-test-interpreter"

# Bin
BIN_FOLDER="bin"

# Icons
ICONS_FOLDER="icons/ios"

# Local game
LOCAL_FOLDER="localgame/ios"

# Ios SDK
IOS_SDK="sdk/iPhoneOS17.5.sdk"
#-platformsdk "$IOS_SDK"

# Lib
LIB_FOLDER="lib/common/"

# Lib2
LIB2_FOLDER="lib/ios/"

# Lib3
LIB3_FOLDER="lib/common_platform/"

## END CONFIG
#####################################

./AIRSDK/AirSdk/bin/adt -package -target "$PACKAGE_COMMAND" -storetype pkcs12 -keystore "$KEY_P12" -storepass "$KEY_PASSWORD" -provisioning-profile "$PROVISIONING_PROFILE" "$IPA_FILE" "$APP_XML" -C "$BIN_FOLDER" . -C "$ICONS_FOLDER" . -C "$LOCAL_FOLDER" . -C "$IOS_SDK". -extdir "$LIB_FOLDER" -extdir "$LIB2_FOLDER" -extdir "$LIB3_FOLDER"

**Comment by Seraphim-AT** [comment]
@a9848840, in AIR config file

[AIR SDK 51.1.2.2 path]/lib/adt.cfg

find commented line

`#iOSPlatformSDK=path_to_sdk` 

uncomment it and add the path to the iOS SDK on your machine, like if you're using MacOS:

`iOSPlatformSDK=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.5.sdk`

**Comment by a9848840** [comment]
hi
@Seraphim-AT 

Thank you, after adding the IOS SDK path according to your method
Packaged successfully
:)

**Comment by marchbold** [comment]
Worth having a look at the AIR SDK Manager to handle this configuration, much easier than editing the cfg file directly.

**Comment by itlancer** [comment]
@ajwfrost 
Original issue still exists with AIR 51.1.2.3 with the same errors in adt.log.

**Comment by itlancer** [comment]
Finally fixed with latest AIR 51.1.3.2 when framework contains another framework.
Thanks!
